### DISCUSSION TITLE
Why doesn't the Alert appear after the Alert?

### DISCUSSION THREAD

**Discussion by ohsg0226** [pointfreeco/swift-composable-architecture#2030] [Q&A]
| "Incremented" Alert not working | "Incremented" Alert working |
|:---:|:---:|
|<img src=https://user-images.githubusercontent.com/65723901/230773868-dc80382a-0164-4211-a50e-380ccbeaac2d.gif width=200 height=300>|<img src=https://user-images.githubusercontent.com/65723901/230774770-0a59d7a0-a325-4319-9014-51dd8435cf87.mp4 width=200 height=300>

First image is 01-GettingStarted-AlertsAndConfirmationDialogs in SwiftUICaseStudies Examples. 
According to the code flow, A new alert should appear when the Incremented button is pressed in the Alert that appears after pressing the Alert button.
But it doesn't work.
I think this problem occurs to "state.alert = nil"
```
func reduce(into state: inout State, action: Action) -> EffectTask<Action> {
    switch action {
...
    case .alertDismissed:
      state.alert = nil
      return .none
...
```
so i deleted  "state.alert = nil"
In this case it works fine like the second image.
But I don't think this is a perfect solution.

I would like some advice to solve this problem.

**Comment by mbrandonw** [comment]
Hi @ohsg0226, thanks for this report, I now see the issue you are describing. Do you know if this works on earlier versions of Xcode/SwiftUI? Have you tested in 14.2, 14.1 or 14? Or maybe even 13.4?

**Reply by ohsg0226** [reply]
hi @mbrandonw 
I tested on version 14.2.
